fairer-iOS/fairer-iOS/Screens/OnboardingName/OnboardingNameViewController.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  OnboardingNameViewController.swift
+//  fairer-iOS
+//
+//  Created by 김유나 on 2022/09/18.
+//
+
+import UIKit
+
+import SnapKit
+
+final class OnboardingNameViewController: BaseViewController {
+    
+    private let nameMaxLength = 5
+    
+    // MARK: - property
+    
+    private let nameLabel: UILabel = {
+        let label = UILabel()
+        label.text = TextLiteral.onboardingNameViewControllerNameLabel
+        label.textColor = .gray800
+        label.font = .h2
+        return label
+    }()
+    private let nameTextField: UITextField = {
+        let textField = UITextField()
+        let attributes = [
+            NSAttributedString.Key.font: UIFont.h3,
+            NSAttributedString.Key.foregroundColor: UIColor.gray400
+        ]
+        textField.backgroundColor = .normal0
+        textField.font = .h3
+        textField.attributedPlaceholder = NSAttributedString(string: TextLiteral.onboardingNameViewControllerTextFieldPlaceholder, attributes: attributes)
+        textField.layer.cornerRadius = 8
+        textField.autocorrectionType = .no
+        textField.autocapitalizationType = .none
+        textField.leftView = UIView(frame: CGRect(x: 0, y: 0, width: 24, height: 58))
+        textField.leftViewMode = .always
+        textField.setClearButton()
+        return textField
+    }()
+    private let nameDoneButton: MainButton = {
+        let button = MainButton()
+        button.title = TextLiteral.onboardingNameViewControllerDoneButtonText
+        button.isDisabled = true
+        button.addTarget(self, action: #selector(didTapDoneButton), for: .touchUpInside)
+        return button
+    }()
+    private let disableLabel: UILabel = {
+        let label = UILabel()
+        label.text = TextLiteral.onboardingNameViewControllerDisableLabel
+        label.textColor = .negative20
+        label.font = .body2
+        label.layer.opacity = 0
+        return label
+    }()
+    
+    // MARK: - life cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupDelegation()
+        setupNotificationCenter()
+    }
+    
+    override func configUI() {
+        super.configUI()
+    }
+    
+    override func render() {
+        view.addSubview(nameLabel)
+        nameLabel.snp.makeConstraints {
+            $0.leading.equalTo(24)
+            $0.top.equalToSuperview().offset(111)
```

Comment:
p3;
```suggestion
            $0.leading.equalToSuperview().inset(24)
            $0.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(10)
```
요렇게 쓸 수 있을 것 같아요 !
nameLabel의 leading은 superview에서 24만큼 떨어진다는 뜻이고, top은 safeLayoutGuide가 상하좌우 다 있는데 그 중에서도 top (현재로선 네비게이션 바로 잡혀요)으로부터 10 만큼 떨어진다는 뜻입니당 !

Reply:
앗 원래 navbar 기준으로 아래 10 offset을 줄까 하다가 제가 baseVC에서 setupNavigationBar 메소드를 호출하지 않아서 일단 superview 기준으로 111 넣었어요! baseVC에서 setupNavigationBar 메소드 호출을 따로 안해도 navbar가 적용되는 걸까요? 제가 VC 따로 빼서 사용을 안해봐서 잘 모르겠네용..

Reply:
아 navbar 까지 safearea 로 생각하는 거군요!
[7923939](https://github.com/fairer-iOS/fairer-iOS/pull/15/commits/79239398bf59d8df61c6b592d82fedb78082b477)
내용 수정해서 반영했습니다 :)

Reply:
SceneDelegate에서 UINavigationController로 출발해서 navigationController가 적용됩니다 !
BaseViewController에서 viewDidLoad에 setupNavigationBar함수가 들어있어서 자동으로 실행됩니다 !

Reply:
아하 따로 override 안해도 실행 되는 군요! 감사해요!